data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/buffer/BufferTopicConfig.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.kafka.buffer;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Size;
+import org.opensearch.dataprepper.model.types.ByteCount;
+import org.opensearch.dataprepper.plugins.kafka.configuration.CommonTopicConfig;
+import org.opensearch.dataprepper.plugins.kafka.configuration.ConsumerTopicConfig;
+import org.opensearch.dataprepper.plugins.kafka.configuration.KafkaKeyMode;
+import org.opensearch.dataprepper.plugins.kafka.configuration.KmsConfig;
+import org.opensearch.dataprepper.plugins.kafka.configuration.ProducerTopicConfig;
+import org.opensearch.dataprepper.plugins.kafka.util.MessageFormat;
+
+import java.time.Duration;
+
+class BufferTopicConfig extends CommonTopicConfig implements ProducerTopicConfig, ConsumerTopicConfig {
+    static final Duration DEFAULT_COMMIT_INTERVAL = Duration.ofSeconds(5);
+    private static final Integer DEFAULT_NUM_OF_PARTITIONS = 1;
+    private static final Short DEFAULT_REPLICATION_FACTOR = 1;
+    private static final Long DEFAULT_RETENTION_PERIOD = 604800000L;
+    static final boolean DEFAULT_AUTO_COMMIT = false;
+    static final ByteCount DEFAULT_FETCH_MAX_BYTES = ByteCount.parse("50mb");
+    static final Duration DEFAULT_FETCH_MAX_WAIT = Duration.ofMillis(500);
+    static final ByteCount DEFAULT_FETCH_MIN_BYTES = ByteCount.parse("1b");
+    static final ByteCount DEFAULT_MAX_PARTITION_FETCH_BYTES = ByteCount.parse("1mb");
+
+
+    @JsonProperty("encryption_key")
+    private String encryptionKey;
+
+    @JsonProperty("kms")
+    private KmsConfig kmsConfig;
+
+    @JsonProperty("commit_interval")
+    @Valid
+    @Size(min = 1)
+    private Duration commitInterval = DEFAULT_COMMIT_INTERVAL;
+
+    @JsonProperty("number_of_partitions")
+    private Integer numberOfPartitions = DEFAULT_NUM_OF_PARTITIONS;
+
+    @JsonProperty("replication_factor")
+    private Short replicationFactor = DEFAULT_REPLICATION_FACTOR;
+
+    @JsonProperty("retention_period")
+    private Long retentionPeriod = DEFAULT_RETENTION_PERIOD;
+
+    @JsonProperty("is_topic_create")
```

Comment:
there is some confusion on meaning of this property. if this is for whether kafka buffer should create topic then may be its should called createTopic.

Reply:
I agree with this comment. I didn't change the name to avoid breaking anything, but this is only used in the sink and buffer, neither of which have been released. So I'll change this to `create_topic`.